src/extension.ts

```diff
@@ -1,21 +1,76 @@
-import * as vscode from 'vscode';
+import { exec } from 'child_process';
+import * as fs from 'fs/promises';
 import * as os from 'os';
 import * as path from 'path';
-import { exec } from 'child_process';
 import { promisify } from 'util';
+import * as vscode from 'vscode';
 import {
+    Executable,
     LanguageClient,
     LanguageClientOptions,
     ServerOptions,
-    TransportKind,
-    Executable
+    TransportKind
 } from 'vscode-languageclient/node';
+import { which } from './which';
 
 let client: LanguageClient;
 
 const execAsync = promisify(exec);
 const MIN_ZIZMOR_VERSION = '1.11.0';
 
+async function getZizmor(config: vscode.WorkspaceConfiguration): Promise<string | undefined> {
+    const rawConfiguredPath = config.get<string>('executablePath');
+
+    if (rawConfiguredPath) {
+        const configuredPath = await which(expandTilde(rawConfiguredPath));
+
+        if (configuredPath) {
+            console.log(`Using configured zizmor: ${configuredPath}`);
+            return configuredPath;
+        } else {
+            console.warn(`Configured zizmor not found: ${rawConfiguredPath}`);
+        }
+    }
+
+    const pathPath = await which('zizmor');
+    if (pathPath) {
+        console.log(`Using zizmor from PATH: ${pathPath}`);
+        return pathPath;
+    }
+
+    // It's particularly important to check common locations on macOS because of https://github.com/microsoft/vscode/issues/30847#issuecomment-420399383.
+    const commonPathsLocal = [
+        path.join(os.homedir(), ".cargo", "bin"),
+        path.join(os.homedir(), ".nix-profile", "bin"),
+        path.join(os.homedir(), ".local", "bin"),
+        path.join(os.homedir(), "bin"),
+    ];
+    const commonPathsUnixGlobal = [
+        "/usr/bin/",
+        "/home/linuxbrew/.linuxbrew/bin/",
+        "/usr/local/bin/",
+        "/opt/homebrew/bin/",
+        "/opt/local/bin/",
+    ];
+
+    // Windows supports forward slashes, but these paths won't work without drive letters.
+    // I figure old DOSes we don't support might not support forward slashes, so this is a good enough check.
+    const commonPaths = path.sep === "/" ? [...commonPathsLocal, ...commonPathsUnixGlobal] : commonPathsLocal;
```

Comment:
Thinking out loud, maybe we can skip all of this for Windows? To my understanding the only reason we need this kind of special casing is because macOS graphical launches don't inherit the login shell environment, so it might be better to just isolate all of the special "common paths" to just macOS (and maybe Linux?) hosts.

Reply:
I don't like special casing at all, so I don't know if special casing the special casing makes it better or worse to me...

It'd be inconsistent, but they're inconsistent anyway, so I guess that makes sense, sure.